src/test/java/model/PlayerTest.java

```diff
@@ -1,29 +1,117 @@
 package model;
 
+
+import exception.InvalidMoneyAmountException;
+import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
-import static org.junit.jupiter.api.Assertions.assertEquals;
-import static org.junit.jupiter.api.Assertions.assertThrows;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
 
 class PlayerTest {
+    Player aPlayer;
+
     @BeforeEach
     void setUp() {
+        aPlayer = new Player();
+        aPlayer.addCardToHand(new Card(Rank.KING, Suit.DIAMOND));
+    }
 
+    @AfterEach
+    void tearDown() {
     }
+
     @Test
-    void withdrawValidAmount() {
+    void addCardToHand() {
+        aPlayer.addCardToHand(new Card(Rank.TWO, Suit.DIAMOND));
+        aPlayer.addCardToHand(new Card(Rank.QUEEN, Suit.DIAMOND));
+        assertEquals(22, aPlayer.getHandValue());
+        List<Card> cardList = aPlayer.hand.getCards();
+        assertTrue(cardList.get(0).equals(new Card(Rank.KING, Suit.DIAMOND)));
+        assertTrue(cardList.get(1).equals(new Card(Rank.TWO, Suit.DIAMOND)));
+        assertTrue(cardList.get(2).equals(new Card(Rank.QUEEN, Suit.DIAMOND)));
+    }
 
+    @Test
+    void balance() {
+        assertEquals(100, aPlayer.balance());
     }
 
     @Test
-    void withdrawTooMuch() {
+    void withdrawValidAmount() {
+        assertDoesNotThrow(() -> aPlayer.withdraw(40));
+        try {
+            aPlayer.withdraw(40);
+            assertEquals(20, aPlayer.balance());
+        } catch (InvalidMoneyAmountException e) {
+            fail();
+        }
+    }
 
+    @Test
+    void withdrawTooMuch() {
+        assertThrows(InvalidMoneyAmountException.class, () -> aPlayer.withdraw(101));
     }
 
     @Test
     void withdraw0() {
+        assertThrows(InvalidMoneyAmountException.class, () -> aPlayer.withdraw(0));
+    }
+
+
+    @Test
+    void receiveMoney() {
+        try {
+            aPlayer.receiveMoney(50);
+        } catch (InvalidMoneyAmountException e) {
+            fail();
+        }
+        assertEquals(150, aPlayer.balance());
+    }
 
+    @Test
+    void receiveMoney0() {
+        assertThrows(InvalidMoneyAmountException.class, () -> aPlayer.receiveMoney(0));
+    }
+
+    @Test
+    void receiveMoneyNegative() {
+        assertThrows(InvalidMoneyAmountException.class, () -> aPlayer.receiveMoney(-50));
+    }
+
+    @Test
+    void getAllHandValues() {
+        assertEquals("    (10)", aPlayer.getAllHandValues());
```

Comment:
This test will fail as soon as the string representation changes slightly (number of white spaces)